### ISSUE TITLE
which parameters are most memory-consuming for CUDA training

### ISSUE THREAD

**Issue by ManoaNosea** [jiny2001/dcscn-super-resolution#59]
what should I reduce for training with lower video memory ?

**Comment by ManoaNosea** [comment]
layers and filters appear to have very litle effect on memory used

**Comment by ManoaNosea** [comment]
are there any possible modifications that could be made to require less in gpu memory ?

**Comment by ManoaNosea** [comment]
--layers and --batch_image_size looks like isn't but --filters appears to be

**Comment by ManoaNosea** [comment]
--nin_filters too

**Comment by jiny2001** [comment]
I believe filters and layers are the most important. But You can reduce batch_num which is the number of images for the one training step.

If you still want to save memory, you can turn gradient clipping and l2 weight decay off like clipping_norm=0 and l2_decay=0. 

Also, the most biggest memory consuming phase would be in the evaluation phase in the training.
Because in the training, input data size would be batch_image_size^2 * batch_num. While in evaluation, input at a would be test_image_width * test_image_height. So you can try to reduce test image's size as well.

**Comment by jiny2001** [comment]
That's not surprising.
I usually use layers=6 and filters=64. When you use not a complicated images for testing, the performance can be like that.

**Comment by ManoaNosea** [comment]
so the size of the NN can be determine by complexity of the data ? this is good news for me :) my card have only 3 giga

**Comment by jiny2001** [comment]
Yep. Because of almost all super resolution model is pixel wise model, input image's resolution will linearly increase the model size as long as they process one image at a time.
